kite-tools-parent/kite-tools/src/main/java/org/kitesdk/cli/commands/DeleteCommand.java

```diff
@@ -46,11 +51,24 @@ public int run() throws IOException {
         View view = Datasets.load(uriOrName);
         Preconditions.checkArgument(viewMatches(view.getUri(), uriOrName),
             "Resolved view does not match requested view: " + view.getUri());
-        view.deleteAll();
+
+        if(console.isDebugEnabled()){
+          for(PartitionView<?> pv: (Iterable<PartitionView<?>>) view.getCoveringPartitions()){
+            console.debug("Deleting partition {}", pv.getLocation());
```

Comment:
Last commit is adding the canDelete method.  Let me know if that is what you were thinking.